test/powershell/engine/Basic/NativeCommandErrorHandling.Tests.ps1

```diff
@@ -48,7 +48,7 @@ Describe 'Native command error handling tests' -Tags 'CI' {
 
             $error[0].FullyQualifiedErrorId | Should -BeExactly 'ProgramExitedWithNonZeroCode'
             $error[0].TargetObject | Should -BeExactly $exePath
-            $stderr[1].Exception.Message | Should -BeExactly "Program `"$exeName`" ended with non-zero exit code: 1."
+            $stderr[1].Exception.Message | Should -BeExactly "Program `"$exeName`" ended with non-zero exit code: 1 (0x00000001)."
```

Comment:
Why not `(0x1)`?

Reply:
Because the hex format used is the fixed-sized `0xNNNNNNNN`, for readability's sake. Call it a matter of taste, but I find reading values such as `0xFE234` "less pretty/readable/usable" than `0x000FE234`.

Reply:
How this lokks in docs https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-

Reply:
Just an FYI while on Windows the rc is a 32-bit integer, on Unix they are just a single byte. I don't have a particular preference either way just wanted to share that in case it helps to make a decision.

Reply:
> Just an FYI while on Windows the rc is a 32-bit integer, on Unix they are just a single byte. I don't have a particular preference either way just wanted to share that in case it helps to make a decision.

Are you somehow suggesting that the parenthesized hex value should use `0x{1:X}` format on non-Windows platforms and `0x{1:X8}` format on Windows?

Reply:
I say about remove leading zeros.

Reply:
@SteveL-MSFT shouldn't that be discussed / decided upon by the team? Who is supposed to have the final say on this?

Reply:
The Engine WG discussed this today and agree that on Linux & macOS, the hexadecimal format should be `0x{1:X2}`.  We also suggest removing the formatting string from the resx and put it in code. That is change this:
```xml
<value>Program "{0}" ended with non-zero exit code: {1} (0x{1:X8}).</value>
```
to
```xml
<value>Program "{0}" ended with non-zero exit code: {1} (0x{2}).</value>
```
Then in code, do something like this:
```csharp
                    const string errorId = nameof(CommandBaseStrings.ProgramExitedWithNonZeroCode);
                    const string hexFormatStr = Platform.IsWindows ? "{0:X8}" : "{0:X2}";

                    string errorMsg = StringUtil.Format(
                        CommandBaseStrings.ProgramExitedWithNonZeroCode,
                        NativeCommandName,
                        _nativeProcess.ExitCode,
                        String.Format(CultureInfo.InvariantCulture, hexFormatStr, _nativeProcess.ExitCode));
```